torch/lib/THCS/generic/THCSTensorMath.cu

```diff
@@ -217,21 +217,25 @@ void THCSTensor_(hspmm)(THCState *state, THCSTensor *r_, real alpha, THCSTensor
 void THCSTensor_(spcadd)(THCState *state, THCTensor *r_, THCTensor *dense, real value, THCSTensor *sparse) {
   THCAssertSameGPU(THCSTensor_(checkGPU)(state, 1, 3, sparse, r_, dense));
 
+  const ptrdiff_t nnz = THCSTensor_(nnz)(state, sparse);
+  if (nnz == 0) {
+    THCTensor_(resizeAs)(state, r_, dense);
+    THCTensor_(copy)(state, r_, dense);
```

Comment:
Looks good. My only nit is that `if (tensor == src) return;` is enough to say "nothing to do". Should be good to go once you add these changes